src/content/blog/2022-08-08-new-config-system-part-3.md

```diff
@@ -0,0 +1,144 @@
+
+## Using flat config with the `Linter` class
+
+If you are currently using `Linter` from the `eslint` package, you can enable flat config by setting `configType: "flat"` as an option on the constructor. Here's an example:
+
+```js
+const linter = new Linter({ configType: "flat" });
+
+const messages = linter.verify("new Map()", {
+    languageOptions: {
+        ecmaVersion: 5,
+        sourceType: "script"
+    },
+    rules: {
+        "no-undef": "error"
+    }
+}, "filename.js");
+```
+
+When you pass `configType: "flat"` as an option, `Linter` expects that any configuration objects that are passed to `verify()` will be in flat config format. In this example, the second argument to `verify()` is a flat config object (you can pass either a single object or an array of objects). Any call to `verify()` will assume that the text being linted is a JavaScript file whose filename ends with `.js`, but it's always a good idea to pass in an explicit filename as the third argument.
+
+While this base case works the same regardless of which config system you're using, there are some important differences:
+
+* `defineRule()`, `defineRules()`, and `defineParser()` now throw errors. Runtime plugins (discussed in my previous post) make these methods obsolete.
+* `getRules()` also throws an error. This method would return different data depending on when it was called, so it can't be used with flat config.
+
+## Using flat config with the `ESLint` class
+
+While implementing flat config we discovered that it would be too difficult to create option to switch between config systems like we did for `Linter`. Instead, we created a `FlatESLint` class that encapsulates all of the existing functionality in `ESLint` but uses flat config instead of eslintrc. The `FlatESLint` class is intended only as a preview of functionality; once we switch over to flat config permanently, the current `ESLint` class will be deleted and `FlatESLint` will be renamed to `ESLint`.
+
+For now, you can access `FlatESLint` through the `use-at-your-own-risk` entrypoint, like this:
+
+```js
+// ESM
+import { FlatESLint } from "eslint/use-at-your-own-risk";
+
+// CommonJS
+const { FlatESLint } = require("eslint/use-at-your-own-risk");
+```
+
+After that, you can use `FlatESLint` in the same way as `ESLint`, such as:
+
+```js
+const eslint = new FlatESLint({
+    cwd: originalDir,
+    overrideConfigFile: "other.config.js"
+});
+const results = await eslint.lintText("foo");
+```
+
+As with `Linter`, there are a few differences between `FlatESLint` and `ESLint` worth pointing out:
+
+1. Caching is not yet implemented in `FlatESLint`, so `cache: true` throws an error.
+1. The `useEslintrc` option has been removed. If you want to avoid automatic loading of `eslint.config.js` without specifying an alternate config file, set `overrideConfigFile: true`.
```

Comment:
I'm confused by this. In the code snippet, a string filename is used. Here, boolean `true` is specified. Are both values supported, or only the filenames?

Reply:
Both are supported.

Reply:
Rereading this, that is now obvious. Thanks for clarifying